components/style/values/computed/pointing.rs

```diff
@@ -0,0 +1,142 @@
+///
+/// https://drafts.csswg.org/css-ui/#cursor
+pub use values::specified::pointing::Cursor;
+#[cfg(feature = "gecko")]
+pub use values::specified::pointing::CursorImage;
+#[cfg(feature = "gecko")]
+use values::specified::url::SpecifiedUrl;
+
+impl Cursor {
+    /// Set `cursor` to `auto`
+    #[cfg(feature = "servo")]
+    #[inline]
+    pub fn auto() -> Self {
+        Cursor(CursorKind::Auto)
+    }
+
+    /// Set `cursor` to `auto`
+    #[cfg(feature = "gecko")]
+    #[inline]
+    pub fn auto() -> Self {
+        Self {
+            images: vec![].into_boxed_slice(),
+            keyword: CursorKind::Auto
+        }
+    }
+}
+
+impl Parse for Cursor {
+    /// cursor: [auto | default | ...]
+    #[cfg(feature = "servo")]
+    fn parse<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<Self, ParseError<'i>> {
+        Ok(Cursor(CursorKind::parse(context, input)?))
+    }
+
+    /// cursor: [<url> [<number> <number>]?]# [auto | default | ...]
+    #[cfg(feature = "gecko")]
+    fn parse<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<Self, ParseError<'i>> {
+        let mut images = vec![];
+        loop {
+            match input.try(|input| CursorImage::parse_image(context, input)) {
+                Ok(mut image) => {
+                    image.url.build_image_value();
+                    images.push(image)
+                }
+                Err(_) => break,
+            }
+            input.expect_comma()?;
+        }
+        Ok(Self {
+            images: images.into_boxed_slice(),
+            keyword: CursorKind::parse(context, input)?,
+        })
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl ToCss for Cursor {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result
+        where W: fmt::Write
+    {
+        for url in &*self.images {
+            url.to_css(dest)?;
+            dest.write_str(", ")?;
+        }
+        self.keyword.to_css(dest)
+    }
+}
+
+impl Parse for CursorKind {
+    fn parse<'i, 't>(
+        _context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<Self, ParseError<'i>> {
+        #[allow(unused_imports)] use std::ascii::AsciiExt;
+        let location = input.current_source_location();
+        let ident = input.expect_ident()?;
+        if ident.eq_ignore_ascii_case("auto") {
+            Ok(CursorKind::Auto)
+        } else {
+            CursorKind::from_css_keyword(&ident)
+                .map_err(|_| location.new_custom_error(
+                        SelectorParseErrorKind::UnexpectedIdent(ident.clone())))
+        }
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl CursorImage {
+    fn parse_image<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>
+    ) -> Result<Self, ParseError<'i>> {
+        Ok(Self {
+            url: SpecifiedUrl::parse(context, input)?,
+            hotspot: match input.try(|input| input.expect_number()) {
```

Comment:
This probably should use `Number::parse`, to handle `calc()`, but let's do this in a followup, since it also needs tests and all that.

But could you leave a comment like:

```rust
// FIXME(emilio): Should use Number::parse to handle calc() correctly.
```

right above `hotspot:`?

Reply:
Sure, done! :)